src/Sentry.Unity/Integrations/TraceGenerationIntegration.cs

```diff
@@ -0,0 +1,39 @@
+using Sentry.Extensibility;
+using Sentry.Integrations;
+
+namespace Sentry.Unity.Integrations;
+
+internal class TraceGenerationIntegration : ISdkIntegration
+{
+    private readonly ISceneManager _sceneManager;
+    private readonly ISentryMonoBehaviour _sentryMonoBehaviour;
+
+    public TraceGenerationIntegration(SentryMonoBehaviour sentryMonoBehaviour) : this(sentryMonoBehaviour, SceneManagerAdapter.Instance)
+    { }
+
+    internal TraceGenerationIntegration(ISentryMonoBehaviour sentryMonoBehaviour, ISceneManager sceneManager)
+    {
+        _sceneManager = sceneManager;
+        _sentryMonoBehaviour = sentryMonoBehaviour;
+    }
+
+    public void Register(IHub hub, SentryOptions options)
+    {
+        hub.ConfigureScope(scope =>
```

Comment:
question: Before this update, I was wondering if we may run into a timing issue when the `ActiveSceneChanged` has already been raised (or is never going to be raised if no scene is ever going to change for the game ... I am unfamiliar with Unity's lifetime events) before we are going to _Register_ this _SDK Integration_.

Is this line covering above scenario, or am I getting something wrong?

Reply:
The `ActiveSceneChanged` might or might not be called many times over the game's lifetime. It is getting called at least once, during startup.

In case of user initializing themselves, we're "missing" out on the first `sceneChange` but then the call to `Register` covers that, making sure that there's a proper trace present.